File: EEntityCore.DB/EEntityCore.VB.MSSQL.WebTest/DBEntities/DatabaseSchema/DatabaseInit.vb
Imports CODERiT.Logger.Standard.VB
Imports EEntityCore.DB
Imports EEntityCore.DB.MSSQL
Imports EEntityCore.DB.MSSQL.Interfaces

Namespace DBEntities.DatabaseSchema

    ''' <summary>
    ''' Make sure you initialize this in your app as DatabaseInit.DBConnectInterface = new DatabaseInit(mode). This class only points to your connection. You can override it
    ''' </summary>
    ''' <remarks></remarks>
    Public Class DatabaseInit
        Implements ISQLServerDBPointer, IDisposable




        REM New Discovery
        REM If Initial Catalog is not specified in connection string, it connects as master


#Region "Constructors"


        Sub New(
                pParent As IDatabaseInit)

            Me.Parent = pParent
        End Sub


#End Region




#Region "Enums and Consts"




#End Region


#Region "Properties"

        Private Parent As IDatabaseInit

        ''' <summary>
        ''' You are responsible for initializing this
        ''' </summary>
        ''' <remarks></remarks>
        Public Shared DBConnectInterface As DatabaseInit



        ''' <summary>
        ''' All usage of DBConn must handle exception
        ''' </summary>
        ''' <value></value>
        ''' <returns></returns>
        ''' <remarks></remarks>
        Private ReadOnly Property DBConnClientRemote As Abstracts.All__DBs
            Get

                Return New Client(Me.Parent.ServerIPAddressOrName, Me.Parent.ServerPort,
                                    Me.Parent.DBUserName, Me.Parent.DBUserPassword, Me.Parent.DBName
                                    )


            End Get
        End Property


        Public Shared ReadOnly Property Logger As Log1
            Get
                Return Server.Logger
            End Get
        End Property


#End Region



#Region "Methods"

        Public Function getDBConn() As Server Implements ISQLServerDBPointer.getDBConn
            Return CType(DBConnClientRemote, Server)
        End Function

#End Region




#Region "IDisposable Support"
        Private disposedValue As Boolean ' To detect redundant calls

        ' IDisposable
        Protected Overridable Sub Dispose(disposing As Boolean)
            If Not Me.disposedValue Then
                If disposing Then
                    ' TODO: dispose managed state (managed objects).
                End If

                ' TODO: free unmanaged resources (unmanaged objects) and override Finalize() below.
                ' TODO: set large fields to null.
            End If
            Me.disposedValue = True
        End Sub

        ' TODO: override Finalize() only if Dispose(ByVal disposing As Boolean) above has code to free unmanaged resources.
        'Protected Overrides Sub Finalize()
        '    ' Do not change this code.  Put cleanup code in Dispose(ByVal disposing As Boolean) above.
        '    Dispose(False)
        '    MyBase.Finalize()
        'End Sub

        ' This code added by Visual Basic to correctly implement the disposable pattern.
        Public Sub Dispose() Implements IDisposable.Dispose
            ' Do not change this code.  Put cleanup code in Dispose(disposing As Boolean) above.
            Dispose(True)
            GC.SuppressFinalize(Me)
        End Sub
#End Region


    End Class

End Namespace

File: EEntityCore.DB/EEntityCore.MSSQL.WebTest/DBEntities/DatabaseSchema/DatabaseInit.cs
using EEntityCore.DB.Interfaces;
using EEntityCore.DB.MSSQL;
using EEntityCore.DB.Schemas.SQLServerSchema;
using System;

namespace EEntityCore.MSSQL.WebTest.DBEntities.DatabaseSchema
{

    /// <summary>
    /// Make sure you initialize this in your app as DatabaseInit.DBConnectInterface = new DatabaseInit(mode). This class only points to your connection. You can override it
    /// DatabaseInit.DBConnectInterface = new DatabaseInit("SalesSCADWARE", "sa", "netEPRO@2017", 51391, @"localhost\SQLSERVER2016");
    /// </summary>
    /// <remarks></remarks>
    public class DatabaseInit : IDatabaseInit
    {

        #region Constructors

        /// <summary>
        /// Just for pointer access
        /// </summary>
        public DatabaseInit()
        {

        }

        /// <summary>
        /// Create Default. NOTE: Server name is not required for FULL_SERVER MODE.
        /// </summary>
        /// <param name="pServerName">with intance name if any like PC\sqlexpress for client modes but for server mode [ Just the SQL Server InstanceName like sqlserver2014 ]</param>
        /// <param name="pDatabaseName">This is very important. the name of the database you are connecting to like [Master]</param>
        /// <param name="pUserName">this is not needed for server mode if server is not allowing clients connection</param>
        /// <param name="pUserPassword">this is not needed for server mode if server is not allowing clients connection</param>
        /// <param name="pPort">this is not needed for server mode if server is not allowing clients connection</param>
        /// <remarks></remarks>
        public DatabaseInit(string pDatabaseName, string pUserName = "", string pUserPassword = "", int pPort = 1433, string pServerName = "")
        {
            ServerName = pServerName;
            UserName = pUserName;
            UserPassword = pUserPassword;
            DatabaseName = pDatabaseName;
            Port = pPort;
        }


        #endregion



        #region Properties

        //private DatabaseSchema.IDatabaseInit Parent;

        /// <summary>
        /// You are responsible for initializing this
        /// </summary>
        /// <remarks></remarks>
        public static DatabaseInit DBConnectInterface;


        private readonly string ServerName;
        private readonly string DatabaseName;
        private readonly string UserName;
        private readonly string UserPassword;
        private readonly int Port;


        #endregion



        #region Methods

        public IDatabaseInit GetDatabaseInit()
        {
            return DBConnectInterface;
        }

        /// <summary>
        /// Returns new connection on each call
        /// </summary>
        /// <returns></returns>
        public Server GetDBConn()
        {
            return (Server)this.GetAllDBConn();
        }

        public IAll__DBs GetAllDBConn()
        {
            return new Client(ServerIPAddressOrName(), ServerPort(), DBUserName(), DBUserPassword(), DBName());
        }

        public string DBName()
        {
            return DatabaseName;
        }

        public string DBUserName()
        {
            return UserName;
        }

        public string DBUserPassword()
        {
            return UserPassword;
        }

        public string ServerIPAddressOrName()
        {
            return ServerName;
        }

        public int ServerPort()
        {
            return Port;
        }




        #endregion


    }
}